wadas_webserver/wadas_webserver_main.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import base64
+import logging
+import os
+import signal
+import socket
+
+from utils import cert_gen, setup_logger
+
+from wadas_webserver.config import CERT_FILEPATH, CERT_FOLDER, KEY_FILEPATH
+from wadas_webserver.database import Database
+from wadas_webserver.web_server import WebServer
+from wadas_webserver.web_server_app import app
+
+flag_run = True
+webserver = None
+
+logger = logging.getLogger(__name__)
+
+
+def handle_shutdown():
+    """Method to catch the SIGINT and SIGTERM signals to properly shut down the process"""
+    global flag_run, webserver
+    logger.info("Killing WADAS web server")
+    stop_server()
+    flag_run = False
+
+
+def stop_server():
+    """Method to properly handle the FastAPI server shutdown"""
+    if webserver:
+        webserver.server.should_exit = True
+
+
+def start_web_server():
+    """Method to start the WADAS FastAPI web server on a separate thread
+    N.B. HTTPS certificates are built on-the-fly and stored under CERT_FOLDER
+    """
+
+    if not os.path.exists(CERT_FOLDER):
+        os.makedirs(os.path.join(CERT_FOLDER))
+        cert_gen(KEY_FILEPATH, CERT_FILEPATH)
+    elif not os.path.exists(CERT_FILEPATH) or not os.path.exists(KEY_FILEPATH):
+        cert_gen(KEY_FILEPATH, CERT_FILEPATH)
+
+    app.server = WebServer("0.0.0.0", 443, CERT_FILEPATH, KEY_FILEPATH)
+    app.server.run()
+    return app.server
+
+
+def blocking_socket():
+    """Method to instantiate a blocking socket on a fixed port
+    to wait communication attempts from WADAS main process
+    """
+    global flag_run
```

Comment:
This can be removed, because you don't change flag_run in the below code (you only read it).